From c4442566dc52fcde677b8d876894759f8ff10e65 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
Date: Thu, 14 Jan 2010 09:38:04 -0200
Subject: [PATCH 6/8] virtio-serial-bus: Add ability to hot-unplug ports

RH-Author: Amit Shah <amit.shah@redhat.com>
Message-id: <1263461886-32077-7-git-send-email-amit.shah@redhat.com>
Patchwork-id: 6352
O-Subject: [RHEL 6 PATCH v4 6/8] virtio-serial-bus: Add ability to hot-unplug
	ports
Bugzilla: 543825
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Amit Shah <amit.shah@redhat.com>
---
 hw/virtio-serial-bus.c |    2 ++
 hw/virtio-serial.h     |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/virtio-serial-bus.c |    2 ++
 hw/virtio-serial.h     |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/virtio-serial-bus.c b/hw/virtio-serial-bus.c
index 207a7fb..7e69ffd 100644
--- a/hw/virtio-serial-bus.c
+++ b/hw/virtio-serial-bus.c
@@ -678,6 +678,8 @@ static int virtser_port_qdev_exit(DeviceState *qdev)
     VirtIOSerialPort *port = DO_UPCAST(VirtIOSerialPort, dev, &dev->qdev);
     VirtIOSerial *vser = port->vser;
 
+    send_control_event(port, VIRTIO_CONSOLE_PORT_REMOVE, 1);
+
     /*
      * Don't decrement nr_ports here; thus we keep a linearly
      * increasing port id. Not utilising an id again saves us a couple
diff --git a/hw/virtio-serial.h b/hw/virtio-serial.h
index ea87b7d..987835d 100644
--- a/hw/virtio-serial.h
+++ b/hw/virtio-serial.h
@@ -52,6 +52,7 @@ struct virtio_console_control {
 #define VIRTIO_CONSOLE_PORT_OPEN	3
 #define VIRTIO_CONSOLE_PORT_NAME	4
 #define VIRTIO_CONSOLE_THROTTLE_PORT	5
+#define VIRTIO_CONSOLE_PORT_REMOVE	6
 
 /* == In-qemu interface == */
 
-- 
1.6.3.rc4.29.g8146

